docs/raii.md

```diff
@@ -0,0 +1,356 @@
+
+## Examples
+The following files have been added to the terra testsuite:
+
+| File Name                     | Description                                                                   |
+|-------------------------------|-------------------------------------------------------------------------------|
+| raii-compose.t                | Tests composition of RAII objects.                                            |
+| raii-copy-generation.t        | Verifies automatic generation of copy methods in RAII classes.                |
+| raii-copy-vs-move-arrays.t    | Verifies copy and move semantics of RAII-array objects.                       |
+| raii-copy-vs-move.t           | Verifies copy and move semantics of RAII objects.                             |
+| raii-copyctr-cast.t           | Tests casting behavior in RAII copy constructors.                             |
+| raii-copyctr.t                | Validates behavior of explicitly defined RAII copy constructors.              |
+| raii-dtor-generation.t        | Checks automatic generation of destructors for RAII objects.                  |
+| raii-dtor.t                   | Tests explicit destructor implementation for RAII resource cleanup.           |
+| raii-init-generation.t        | Verifies automatic generation of initializers for RAII objects.               |
+| raii-initializers.t           | Tests explicit initializer implementation in RAII classes.                    |
+| raii-integration-copy.t       | Integration tests of RAII classes with focus on value semantics.              |
+| raii-integration-move.t       | Integration tests of RAII classes with focus on move semantics.               |
+| raii-meta.t                   | Tests use of RAII objects in macros.                                          |
+| raii-shared_ptr.t             | Validates RAII with `shared_ptr` for shared resource ownership.               |
+| raii-unique_ptr.t             | Tests RAII with `unique_ptr` for unique resource ownership.                   |
+| raii-offset_ptr.t             | Tests RAII with offset-based pointer implementations.                         |
+| raii.t                        | General tests for core RAII principles and functionality.                     |
+| fails/raii-tuple-default-copy.t | Checks if tuple assignment is prohibited for managed moveable types         |
+| fails/raii-tuple-custom-copy.t | Checks if tuple assignment is prohibited for managed copyable types          |
+
+You can have a look there for some common code patterns. Useful, in particular, are the integration tests in 'raii-integration-copy.t' and 'raii-integration-move.t'.
+
+## Current limitations
+* The implementation is not aware of when an actual heap allocation is made and therefore assumes that a managed variable always carries a heap resource. It is up to the programmer to properly initialize pointer variables to nil to avoid calling 'free' on uninitialized pointers.
```

Comment:
```suggestion
* The implementation is not aware of when an actual heap allocation is made and therefore assumes that a managed variable always carries a heap resource. It is up to the programmer to properly initialize pointer variables to `nil` to avoid calling `free` on uninitialized pointers.
```